packages/open-next/src/adapters/server-adapter.ts

```diff
@@ -131,9 +131,22 @@ export async function handler(
 
   fixCacheHeaderForHtmlPages(internalEvent.rawPath, headers);
   fixSWRCacheHeader(headers);
+
+  // we are sending a revalidation request that will bypass the middleware https://nextjs.org/docs/pages/building-your-application/data-fetching/incremental-static-regeneration#on-demand-revalidation
+  // so we need to use the rewritten url if it was rewritten
+  let revalidateUrl = internalEvent.rawPath;
+  if (req[Symbol.for("NextInternalRequestMeta")]?._nextDidRewrite) {
+    const rewrittenUrl =
+      req[Symbol.for("NextInternalRequestMeta")]?._nextRewroteUrl;
+
+    // the rewritten url will does not include the /_next/data/ prefix so we need to add it back
+    revalidateUrl = internalEvent.rawPath.startsWith("/_next/data/")
```

Comment:
I'm confused by the comment and the code. Shouldn't it be `!internalEvent.rawPath...`? ie if it doesn't start w/ `/_next/data/`, the prepend it.

Reply:
No so basically from my testing we have two separate events / requests for each page with ISR:
1. First request we have `internalEvent.rawPath === '/some-path'` and `rewrittenUrl === '/some-path-rewritten'`
2. 2nd Request  we have `internalEvent.rawPath === '/_next/data/buildId/some-path.json'` but the rewrittenUrl is still the same and does not include the _next/data/ thing: `rewrittenUrl === '/some-path-rewritten'`. We could potentially send both requests to validate directly when we receive the first request but to make it consistent with the way things work without revalidation I kept is like that.
I probably can add more comments there to make it more easily understandable

Reply:
gotcha, thanks for the update.